public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -234,8 +234,12 @@ export class ElasticDatasource {
       ignore_unavailable: true,
       index: this.indexPattern.getIndexList(timeFrom, timeTo),
     };
-    if (this.esVersion >= 56) {
+    if (this.esVersion >= 56 && this.esVersion < 70) {
       queryHeader['max_concurrent_shard_requests'] = this.maxConcurrentShardRequests;
+    } else if (this.esVersion >= 70) {
+      queryHeader['preference'] = {
```

Comment:
Out of curiosity, where did you find information about this `preference` property change?

Reply:
Hmm, good catch. 

ES 7 will complain about the parameter in the old position, but my edit is only hiding it from the query validator, not doing what I thought it did. 
For that matter,  [it didn't have the expected effect before either, it just didn't complain](https://github.com/elastic/elasticsearch/pull/35938) .

The correct way to pass this parameter is via a [URL query parameter](https://github.com/elastic/elasticsearch/blob/687978b7d167d4c20a29861b2558b2749c349529/server/src/main/java/org/elasticsearch/rest/RestRequest.java#L110), [apparently](https://github.com/elastic/elasticsearch/blob/622fb7883b4e7c6de6dce2636108e334cca28cd3/server/src/main/java/org/elasticsearch/rest/action/search/RestMultiSearchAction.java#L110).

I will try to fix and test this tomorrow.

Reply:
Ok, now it's here: fb8491cca5316620d4d9bbdfae75615596f9afcd

Reply:
Thanks. So frustrating that you actually have to look at the code to understand how their API works.

Reply:
Did some changes. Based on what I can see in the code the search_type is still valid to provide in header of _msearch. Simplified some code and fix max concurrent shard requests in backend.